root/types/tablecomponents.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * @flow strict
+ * Copyright (C) 2017 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+/*
+ * Types are in alphabetical order.
+ *
+ * The definitions in this file are intended to model the output of the
+ * TO_JSON methods under lib/MusicBrainz/Server/Entity/, those are precisely
+ * how data is serialized for us.
+ */
+
+declare type AppearancesT<T> = {
+  +hits: number,
+  +results: $ReadOnlyArray<T>,
+};
```

Comment:
this might make more sense in entity.js since it applies to several entities and I wouldn't associate it with a table (it's used by the autocomplete)